frontend/lib/src/WidgetStateManager.ts

```diff
@@ -837,6 +930,43 @@ export class WidgetStateManager {
       this.elementStates.delete(elementId)
     }
   }
+
+  /**
+   * Schedule a macrotask-level flush of pending widget updates (triggers and
+   * regular value changes). Multiple calls within the same macrotask will be
+   * coalesced into a single `updateWidgets` message.
+   */
+  private scheduleFlush(fragmentId: string | undefined): void {
+    // Update the stored fragmentId if we don't have one yet. If multiple calls
+    // happen and at least one of them specifies a fragmentId, we keep the
+    // first non-undefined value.
```

Comment:
Reasoning through this. What happens if multiple fragment ids are scheduled? Does one get ignored essentially? Could this create a scenario where some widgets get associated with the wrong fragment id?

Reply:
I think (2) feels fine enough if we suspect this should be a very rare use case (and we will then have some artifact from people to detect this.

Just make sure the comment clarifies this agreement that this leaves us exposed to a potential bug but it would need to be really weird setup and we made the decision to just log a warning for now.